File: 1137. N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 1 or n == 2:
            return 1
        t0 = 0
        t1 = 1
        t2 = 1
        t3 = 0
        for i in range(2, n):
            t0 = t1 + t2 + t3
            t1, t2, t3 = t0, t1, t2
        return t0

File: 1137. N-th Tribonacci Number.c
int tribonacci(int n){
    if(n == 0){
        return 0;
    }
    int t0 = 1;
    int t1 = 1;
    int t2 = 1; 
    int t3 = 0;
    for(int i = 2; i < n; i++){
        t0 = t1 + t2 + t3; 
        t3 = t2;
        t2 = t1;
        t1 = t0;
    }
    return t0;
}

//
